Composer/packages/lib/indexers/src/dialogIndexer.ts

```diff
@@ -127,27 +127,26 @@ function ExtractReferredDialogs(dialog): string[] {
 
 // check all fields
 function CheckFields(dialog, id: string, schema: any): Diagnostic[] {
-  const errors: Diagnostic[] = [];
-  const expressionProperties = getExpressionProperties(schema);
+  const diagnostics: Diagnostic[] = [];
   /**
    *
    * @param path , jsonPath string
    * @param value , current node value    *
    * @return boolean, true to stop walk
    * */
   const visitor: VisitorFunc = (path: string, value: any): boolean => {
-    // it's a valid schema dialog node.
-    if (has(value, '$type') && has(expressionProperties, value.$type)) {
-      const diagnostics = IsExpression(path, value, { ...expressionProperties[value.$type] });
-
-      if (diagnostics) {
-        errors.push(...diagnostics);
-      }
+    if (has(value, '$type')) {
+      checkerFuncs.forEach(checkerFunc => {
+        const result = checkerFunc(path, value, value.$type, schema.definitions[value.$type]);
```

Comment:
Here, you are applying every checker on every node regardless of the $type. Then inside a special checker like "checkChoices", you check the $type, and stop if it's not a certain type. 

This works, but very readable, because if you think about we have more and more special checkers later, i have to go into every checker to see how this type of node is getting checked. 

Instead, if i use a map of [type=>checkers or list[checkers]], i would know every precisely, what type of checking is performed for a given type. 

What do you think?